### ISSUE TITLE
Test Helpers in Factory break development build

### ISSUE THREAD

**Issue by patocallaghan** [danielspaniel/ember-data-factory-guy#247]
Since the new development mode has shipped I notice that factories are being included in the development build in the app's JS by default and even when I set `ENV.factoryGuy = false`, i.e. I don't want to use Factory Guy in development mode.

In one of my factories I import a test helper like so:

```
import { getAdminFixture } from 'embercom/tests/helpers/data';
```

but because the factory is being included in dev build and the test helpers aren't, it breaks my app in development mode with the following error: `Could not find module embercom/tests/helpers/data imported from embercom/tests/factories/engage/message-variation`

Should we not be excluding the factories from development builds if you don't want to use the new development mode?

**Comment by danielspaniel** [comment]
Oooo .. you are very right .. those factories should not be in dev if you not using them. 
Great bug by the way. You always find some really interesting ones. 
 I will get my elves to work on that tonight, so you can be back in business tomorrow AM.

**Comment by danielspaniel** [comment]
This was good fix. Should have thought of this before. 

Fixed in https://github.com/danielspaniel/ember-data-factory-guy/releases/tag/v2.7.8

**Comment by GavinJoyce** [comment]
@danielspaniel this commit seems to have reintroduced this issue: https://github.com/danielspaniel/ember-data-factory-guy/commit/f453e9f6ad98adc46cf2a39594c5a85bd2b23ce1

**Comment by danielspaniel** [comment]
I think I was wrong @GavinJoyce to say that the factories should not be included in dev build because they still need to be for the ember test runner which runs when you go to browser and use the /tests route. 
 But maybe I am wrong about that because why would ember test loader run tests and not include all the test files automatically? 
 Can you confirm that I am not loosing my noodle, and that when you run things in /tests url that this is development environment and that all test files/helpers are available. 
 When are you finding this issue? Is it when you run  ember t ( test environment )  ... that would make no sense since all test files are also included .. so I am a bit puzzled.

**Comment by GavinJoyce** [comment]
Thanks @danielspaniel. I'm not fully sure of the original context to this issue, just that it's reoccurring in our app since updating `ember-data-factory-guy`. @patocallaghan has more context, he's going to look into the details and post more info when he has it

**Comment by patocallaghan** [comment]
@danielspaniel we are seeing this issue when we load our app normally in development. We don't want to use FactoryGuy in our local apps.

You are right that FactoryGuy should load if `environment == 'test'` or if hitting `tests/index.html && environment == 'dev'`. But by default FactoryGuy shouldn't be included in `dev`. According to that change above it is always opting the `dev` environment into including FactoryGuy. It's bypassing the `factoryGuyEnabled` config option.

**Comment by danielspaniel** [comment]
Ok @patocallaghan.
 So I think what is causing problem is I am putting the factory guy files into the application tree. 
 For people that want to use factory guy in dev that is fine, but for people that don't I have to put those factory guy files in the test tree ( tests.js ) 
 I recall ( sort of ) that there was a problem with that before but I will try again and see if that works.

**Comment by danielspaniel** [comment]
@patocallaghan  try https://github.com/danielspaniel/ember-data-factory-guy/releases/tag/v2.9.7
 I tested it quite a bit, but I am still nervous. This is squirrelly issue until I can think of how to write some tests to make sure it's spot on.

**Comment by GavinJoyce** [comment]
@danielspaniel thanks, that works in our app

**Comment by danielspaniel** [comment]
Great, glad that worked @GavinJoyce.

**Comment by patocallaghan** [comment]
Thanks @danielspaniel 🙇

**Comment by danielspaniel** [comment]
your welcome @patocallaghan .. glad it's working out ( nice emoji .. me like )

**Comment by danielspaniel** [comment]
hey @patocallaghan .. this issue is returning to haunt me. 

Someone setup a new project with latest factory guy and it was total disaster ( since acceptance tests kept barfing ) .. and I finally figured out it was this issue again. 

 Question for you:
 
Because you have  references to other test files in factories 

 is there a way that you could put all your "helper" files that you import into factories
 into one directory like tests/utils or tests/fg-helpers 
  anything you like really .. 

 because I am needing to go back to this:

```javascript
  includeFactoryGuyFiles: function() {
    var includeFiles = false;
  
    // changed to /test/ to not include in development, but that not working so well 
    if (this.app.env.match(/test|development/)) {
      includeFiles = true;
    } else {
      includeFiles = this.factoryGuyEnabled;
    }

    return includeFiles;
  },
```

since if I don't include the files in development running an acceptance tests with localhost:port/tests  will not load the factories ( since the includeFactoryGuyFiles function is returning false in development mode. ) 
 
  Not sure how this ever worked for anyone frankly.  But the change was in 2.9.7 so maybe not too many people upgraded, or they were using factoryGuy: true in environment.js ( development section ) 

 But need to come up with a solution asap. 

```javascript
  treeForApp: function(appTree) {
    var trees = [appTree];

    if (this.includeFactoryGuyFiles()) {
      try {
        if (fs.statSync('tests/factories').isDirectory()) {
          var factoriesTree = new Funnel('tests/factories', {
            destDir: 'tests/factories'
          });
          var utilsTree = new Funnel('tests/helpers', {  // tests/blah 
            destDir: 'tests/helpers'   // tests/blah 
          });
          trees.push(factoriesTree);
          trees.push(utilsTree);
        }
      } catch (err) {
        // do nothing;
      }
    }

    return mergeTrees(trees);
  },
```

So, basically I need a way to include factory guy still in development, so it works in the /tests mode 
and then also import any files that factories might use. If you can put them in one directory ( suggest me a name ) that would solve the problem.  
 
I can think of no other alternative. Can you? I am open to any and all ideas.

**Comment by patocallaghan** [comment]
@danielspaniel just to confirm that 2.11.0 works for us when we set:

```
ENV.factoryGuy = { enabled: false, useScenarios: false };
```

Factory Guy doesn't appear in our dev environment. Thanks for making those changes 👍

**Comment by danielspaniel** [comment]
Thanks for confirming .. @patocallaghan 😃

**Comment by rreckonerr** [comment]
> ENV.factoryGuy = { enabled: false, useScenarios: false };

For `v2.13.27` dev build works well, but if you navigate to `/tests` most likely you'll find your test suit broken.
Upgraded to `v3.9.7` (latest ATM) and it works good for `/tests` on a dev build  route too.
